backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -995,6 +1027,71 @@ public void legacyRbacUpdateEventTypesLinkedToEndpoint(final SecurityContext sec
 
     private void internalUpdateEventTypesLinkedToEndpoint(final SecurityContext securityContext, final UUID endpointId, final Set<UUID> eventTypeIds) {
         final String orgId = getOrgId(securityContext);
-        endpointEventTypeRepository.updateEventTypesLinkedToEndpoint(endpointId, eventTypeIds, orgId);
+        Endpoint updatedEndpoint =  endpointEventTypeRepository.updateEventTypesLinkedToEndpoint(endpointId, eventTypeIds, orgId);
+
+        // Sync behavior group model
+
+        // delete endpoint from existing behavior group
+        List<BehaviorGroup> behaviorGroupsLinkedToThisEndpoint = behaviorGroupRepository.findBehaviorGroupsByEndpointId(orgId, endpointId);
+        for (BehaviorGroup behaviorGroup : behaviorGroupsLinkedToThisEndpoint) {
+            if (behaviorGroup.getActions().size() == 1) {
+                behaviorGroupRepository.delete(orgId, behaviorGroup.getId());
+            } else {
+                behaviorGroupRepository.deleteEndpointFromBehaviorGroup(behaviorGroup.getId(), endpointId, orgId);
+            }
+        }
```

Comment:
This will affect behavior groups created by customers. Should we have a way (boolean field?) to distinguish them from the behavior groups that are internally created because we're transitioning away from behavior groups?